// cloudflare/resources/workers-for-platforms/dispatch/namespaces/scripts/secrets.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../../resource.mjs";
import { SinglePage } from "../../../../../pagination.mjs";
export class Secrets extends APIResource {
    /**
     * Add a secret to a script uploaded to a Workers for Platforms namespace.
     *
     * @example
     * ```ts
     * const secret =
     *   await client.workersForPlatforms.dispatch.namespaces.scripts.secrets.update(
     *     'my-dispatch-namespace',
     *     'this-is_my_script-01',
     *     {
     *       account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *       name: 'myBinding',
     *       text: 'My secret.',
     *       type: 'secret_text',
     *     },
     *   );
     * ```
     */
    update(dispatchNamespace, scriptName, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/workers/dispatch/namespaces/${dispatchNamespace}/scripts/${scriptName}/secrets`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List secrets bound to a script uploaded to a Workers for Platforms namespace.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const secretListResponse of client.workersForPlatforms.dispatch.namespaces.scripts.secrets.list(
     *   'my-dispatch-namespace',
     *   'this-is_my_script-01',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(dispatchNamespace, scriptName, params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/workers/dispatch/namespaces/${dispatchNamespace}/scripts/${scriptName}/secrets`, SecretListResponsesSinglePage, options);
    }
    /**
     * Remove a secret from a script uploaded to a Workers for Platforms namespace.
     *
     * @example
     * ```ts
     * const secret =
     *   await client.workersForPlatforms.dispatch.namespaces.scripts.secrets.delete(
     *     'my-dispatch-namespace',
     *     'this-is_my_script-01',
     *     'mySecret',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(dispatchNamespace, scriptName, secretName, params, options) {
        const { account_id, url_encoded } = params;
        return this._client.delete(`/accounts/${account_id}/workers/dispatch/namespaces/${dispatchNamespace}/scripts/${scriptName}/secrets/${secretName}`, { query: { url_encoded }, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Create, update, or delete multiple secrets on a script in a single operation
     * using JSON Merge Patch (RFC 7396).
     *
     * Usage:
     *
     * - To create or update a secret, set its value to a secret object.
     * - To delete a secret, set its value to `null`.
     * - Secrets not included in the request are left unchanged.
     *
     * @example
     * ```ts
     * const response =
     *   await client.workersForPlatforms.dispatch.namespaces.scripts.secrets.bulkUpdate(
     *     'my-dispatch-namespace',
     *     'this-is_my_script-01',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    bulkUpdate(dispatchNamespace, scriptName, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/workers/dispatch/namespaces/${dispatchNamespace}/scripts/${scriptName}/secrets-bulk`, {
            body,
            ...options,
            headers: { 'Content-Type': 'application/merge-patch+json', ...options?.headers },
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get a given secret binding (value omitted) on a script uploaded to a Workers for
     * Platforms namespace.
     *
     * @example
     * ```ts
     * const secret =
     *   await client.workersForPlatforms.dispatch.namespaces.scripts.secrets.get(
     *     'my-dispatch-namespace',
     *     'this-is_my_script-01',
     *     'mySecret',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(dispatchNamespace, scriptName, secretName, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/workers/dispatch/namespaces/${dispatchNamespace}/scripts/${scriptName}/secrets/${secretName}`, { query, ...options })._thenUnwrap((obj) => obj.result);
    }
}
export class SecretListResponsesSinglePage extends SinglePage {
}
Secrets.SecretListResponsesSinglePage = SecretListResponsesSinglePage;
