// cloudflare/resources/r2/buckets/objects.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectListResponsesCursorPagination = exports.Objects = void 0;
const resource_1 = require("../../../resource.js");
const pagination_1 = require("../../../pagination.js");
class Objects extends resource_1.APIResource {
    /**
     * Lists objects in an R2 bucket. Returns object metadata including key, size,
     * etag, last modified date, HTTP metadata, and custom metadata.
     *
     * For most workloads, we recommend using R2's
     * [S3-compatible API](https://developers.cloudflare.com/r2/api/s3/api/) or a
     * [Worker with an R2 binding](https://developers.cloudflare.com/r2/api/workers/workers-api-reference/)
     * instead.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const objectListResponse of client.r2.buckets.objects.list(
     *   'example-bucket',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(bucketName, params, options) {
        const { account_id, jurisdiction, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/r2/buckets/${bucketName}/objects`, ObjectListResponsesCursorPagination, {
            query,
            ...options,
            headers: {
                ...(jurisdiction?.toString() != null ?
                    { 'cf-r2-jurisdiction': jurisdiction?.toString() }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    /**
     * Deletes an object from an R2 bucket.
     *
     * For most workloads, we recommend using R2's
     * [S3-compatible API](https://developers.cloudflare.com/r2/api/s3/api/) or a
     * [Worker with an R2 binding](https://developers.cloudflare.com/r2/api/workers/workers-api-reference/)
     * instead.
     *
     * @example
     * ```ts
     * const object = await client.r2.buckets.objects.delete(
     *   'example-bucket',
     *   'path/to/my-object.txt',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(bucketName, objectKey, params, options) {
        const { account_id, jurisdiction } = params;
        return this._client.delete(`/accounts/${account_id}/r2/buckets/${bucketName}/objects/${objectKey}`, {
            ...options,
            headers: {
                ...(jurisdiction?.toString() != null ?
                    { 'cf-r2-jurisdiction': jurisdiction?.toString() }
                    : undefined),
                ...options?.headers,
            },
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Retrieves an object from an R2 bucket. Returns the object body along with
     * metadata headers.
     *
     * For most workloads, we recommend using R2's
     * [S3-compatible API](https://developers.cloudflare.com/r2/api/s3/api/) or a
     * [Worker with an R2 binding](https://developers.cloudflare.com/r2/api/workers/workers-api-reference/)
     * instead.
     *
     * @example
     * ```ts
     * const object = await client.r2.buckets.objects.get(
     *   'example-bucket',
     *   'path/to/my-object.txt',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     *
     * const content = await object.blob();
     * console.log(content);
     * ```
     */
    get(bucketName, objectKey, params, options) {
        const { account_id, jurisdiction, 'If-Modified-Since': ifModifiedSince, 'If-None-Match': ifNoneMatch, } = params;
        return this._client.get(`/accounts/${account_id}/r2/buckets/${bucketName}/objects/${objectKey}`, {
            ...options,
            headers: {
                Accept: 'application/octet-stream',
                ...(jurisdiction?.toString() != null ?
                    { 'cf-r2-jurisdiction': jurisdiction?.toString() }
                    : undefined),
                ...(ifModifiedSince != null ? { 'If-Modified-Since': ifModifiedSince } : undefined),
                ...(ifNoneMatch != null ? { 'If-None-Match': ifNoneMatch } : undefined),
                ...options?.headers,
            },
            __binaryResponse: true,
        });
    }
    /**
     * Uploads an object to an R2 bucket. The object body is provided as the request
     * body. Returns metadata about the uploaded object.
     *
     * The maximum upload size for this endpoint is 300 MB. For most workloads, we
     * recommend using R2's
     * [S3-compatible API](https://developers.cloudflare.com/r2/api/s3/api/) or a
     * [Worker with an R2 binding](https://developers.cloudflare.com/r2/api/workers/workers-api-reference/)
     * instead.
     *
     * @example
     * ```ts
     * const response = await client.r2.buckets.objects.upload(
     *   'example-bucket',
     *   'path/to/my-object.txt',
     *   fs.createReadStream('path/to/file'),
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    upload(bucketName, objectKey, body, params, options) {
        const { account_id, jurisdiction, 'cf-r2-storage-class': cfR2StorageClass } = params;
        return this._client.put(`/accounts/${account_id}/r2/buckets/${bucketName}/objects/${objectKey}`, {
            body: body,
            ...options,
            headers: {
                'Content-Type': 'application/octet-stream',
                ...(jurisdiction?.toString() != null ?
                    { 'cf-r2-jurisdiction': jurisdiction?.toString() }
                    : undefined),
                ...(cfR2StorageClass?.toString() != null ?
                    { 'cf-r2-storage-class': cfR2StorageClass?.toString() }
                    : undefined),
                ...options?.headers,
            },
            __binaryRequest: true,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Objects = Objects;
class ObjectListResponsesCursorPagination extends pagination_1.CursorPagination {
}
exports.ObjectListResponsesCursorPagination = ObjectListResponsesCursorPagination;
Objects.ObjectListResponsesCursorPagination = ObjectListResponsesCursorPagination;
