// cloudflare/resources/dls/regional-services/prefix-bindings.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import { CursorPagination } from "../../../pagination.mjs";
export class PrefixBindings extends APIResource {
    /**
     * Create a DLS prefix binding
     *
     * @example
     * ```ts
     * const prefixBinding =
     *   await client.dls.regionalServices.prefixBindings.create({
     *     account_id: 0,
     *     cidr: '10.0.1.0/24',
     *     prefix_id: 'a1b2c3d4-e5f6-7890-abcd-ef1234567890',
     *     region_key: 'eu',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/dls/regional_services/prefix_bindings`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List DLS prefix bindings for an account
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const prefixBindingListResponse of client.dls.regionalServices.prefixBindings.list(
     *   { account_id: 0 },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/dls/regional_services/prefix_bindings`, PrefixBindingListResponsesCursorPagination, { query, ...options });
    }
    /**
     * Delete a DLS prefix binding
     *
     * @example
     * ```ts
     * const prefixBinding =
     *   await client.dls.regionalServices.prefixBindings.delete(
     *     'a1b2c3d4-e5f6-7890-abcd-ef1234567890',
     *     { account_id: 0 },
     *   );
     * ```
     */
    delete(bindingId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/dls/regional_services/prefix_bindings/${bindingId}`, options);
    }
    /**
     * Update a DLS prefix binding
     *
     * @example
     * ```ts
     * const response =
     *   await client.dls.regionalServices.prefixBindings.edit(
     *     'a1b2c3d4-e5f6-7890-abcd-ef1234567890',
     *     { account_id: 0, region_key: 'eu' },
     *   );
     * ```
     */
    edit(bindingId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/dls/regional_services/prefix_bindings/${bindingId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get a DLS prefix binding
     *
     * @example
     * ```ts
     * const prefixBinding =
     *   await client.dls.regionalServices.prefixBindings.get(
     *     'a1b2c3d4-e5f6-7890-abcd-ef1234567890',
     *     { account_id: 0 },
     *   );
     * ```
     */
    get(bindingId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/dls/regional_services/prefix_bindings/${bindingId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
export class PrefixBindingListResponsesCursorPagination extends CursorPagination {
}
PrefixBindings.PrefixBindingListResponsesCursorPagination = PrefixBindingListResponsesCursorPagination;
